### ISSUE TITLE
Cityscapes + Deeplabv3 benchmark

### ISSUE THREAD

**Issue by Landanjs** [mosaicml/composer#67]
## 🚀 Feature Request

Add a semantic segmentation benchmark based on the Cityscapes dataset and the Deeplabv3 architecture.

### Prior work

Our current segmentation benchmark is based on the Multimodal Brain Tumor Segmentation Challenge (BraTS) and the Unet architecture. There are a couple of reasons why we may want to add another segmentation benchmark:

1. Dataset size: BraTS has lower resolution (192 x 160) and a smaller number of training images (500) than we expect from other segmentation datasets. As of now, we can train on BraTS in ~3 minutes
2. Recognition: BraTS does not seem as recognizable in the ML community, so it may be difficult for people to interpret our results. Also, a frequently used dataset would be beneficial to the community since proposed methods can be easily compared to prior work using the dataset.
3. Domain: this may just be a me thing, but I think it would be helpful to have a dataset in a similar domain to the ImageNet benchmark. This could help in determining whether the success/failure of a method depends on the task or the domain.

### Cityscapes

Cityscapes appears to be the second most common semantic segmentation benchmark (behind Pascal VOC), so evaluating methods on Cityscapes should be relevant to the community. Cityscapes image resolution is 1024 x 2048 and the training set contains 2,975 densely and 20,000 coarsely annotated images (not as many as we would like, but a start). Alternatively, we could use ADE20k or Pascal VOC segmentation if others feel strongly towards either dataset.

### Deeplabv3

It would be easier to benchmark with Deeplabv3 since the hyperparameters and target performance on Cityscapes are known. As of now, we have no numbers on training time, so this will be unknown.  For Unet, we would need to tune hyperparameters and would not be sure if we are achieving an expected performance. 

## Implementation

Simple implementation outline, but should be made more detailed: 

- [ ] Cityscapes DataloaderSpec: will try to use [torchvision.datasets.cityscapes](https://pytorch.org/vision/stable/datasets.html#cityscapes) if it fits our use case

- [ ] Deeplabv3 BaseMosaicModel:  will try to use [torchvision.models.segmentation.deeplabv3_resnet101](https://pytorch.org/vision/stable/models.html#deeplabv3) if it fits our use case

- [ ] Implement intersection-over-union (IoU) metric for evaluation: should be easyish?

- [ ] Dataset and model throughput profiling

- [ ] Dataset and model card
